src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/ObjectFormatting/Formatters/RuntimeFormatters.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.ObjectFormatting
+{
+    internal static class RuntimeFormatters
+    {
+        public static string IConvertibleFormatter(object obj, FormatSpecifier formatSpecifier)
+        {
+            if (obj is not string)
+            {
+                formatSpecifier |= FormatSpecifier.NoQuotes;
+            }
+
+            string formatted = ((IConvertible)obj).ToString(CultureInfo.InvariantCulture);
```

Comment:
For this and IFormattableFormatter, could you check the cast and throw a more informative exception? e.g.

```
if (obj is not IConvertable convertable)
{
    throw new ArgumentException();
}
```

Better yet, change the parameter to be IConvertable and have the caller test using this casting methodology rather than checking with IsAssignableTo/From.

Reply:
Ignore the second part, I see the factory is provided the type and not the actual object.